### DISCUSSION TITLE
ChatGPT in `package.json`? gpt-generate command usage

### DISCUSSION THREAD

**Discussion by leovigna** [TanStack/router#1243] [Q&A] [answered]
## TLDR
* What is the purpose of the `gpt-generate` command?
* How are you meant to use the generated output? Are there any specific plugins you recommend? GithubCopilot or other ones?
* How has your experience been using whatever solution the `db.json` output is fed to?

## Exploring a GPT-integrated package
I recently cloned Tanstack/router to explore the repo in more detail as I have been an avid user of the library.

While exploring the `package.json` I came across the `gpt-generate` command which runs [gpt/generate.js](https://github.com/TanStack/router/blob/main/gpt/generate.js). This command seems to generate a `db.json` file with code source content and docs. This is the first time I see this in a repo so was wondering how this works and its purpose. I checked out the commit that created the file but could not find any more context.

I'm curious if using the `db.json` output with some AI assistant can help me learn more about Tanstack and if I could potentially use a similar pattern in my own codebase.

Appreciate any comments or suggestions!

**Comment by tannerlinsley** [comment, marked as answer]
I use it to generate this GPT bot: https://chat.openai.com/g/g-0FkAVdIHq-tanstack-router-bot

it’s as simple as uploading the output to the bot config as a knowledge source.

**Reply by leovigna** [reply]
Thanks for quick reply!
